src/cnaas_nms/tools/security.py

```diff
@@ -18,6 +27,175 @@ def jwt_required(fn):
 def get_jwt_identity():
     """
     This function overides the identity when needed.
-
     """
     return get_jwt_identity_orig() if api_settings.JWT_ENABLED else "admin"
+
+
+def get_oauth_userinfo(token_string):
+    """Give back the user info of the OAUTH account
+
+    If JWT is disabled, we return "admin".
+
+    We do an api call to request userinfo. This gives back all the userinfo.
+    We get the right info from there and return this to the user.
+
+    Returns:
+        resp.json(): Object of the user info 
+
+    """
+    # For now unnecersary, useful when we nly use one log in method
+    if not auth_settings.OIDC_ENABLED:
+        return "Admin"
+    # Request the userinfo
+    metadata = requests.get(auth_settings.OIDC_CONF_WELL_KNOWN_URL)
```

Comment:
Sorry, one more error that I think it would be good if we catch, if the metadata is unavailable when doing any API request the client will get back message: "Internal server error", if we catch requests.exceptions.ConnectionError here and return something like "OIDC metadata unavailable: ConnectionError" (or whatever error from str(e) )

Reply:
I had troubles reproducing the error, so I had troubles testing it. I catched the error you mentoined and another error I got instead. Let me know how you produced this error if the current changes are not catching the error you meant.